templates/create-daml-app/ui/src/components/MainScreen.tsx

```diff
@@ -14,6 +15,15 @@ type Props = {
  * React component for the main screen of the `App`.
  */
 const MainScreen: React.FC<Props> = ({onLogout}) => {
+  const ledger = useLedger();
+  const [user, setUser] = useState('');
+  useEffect( () =>{
+    (async () => {
+      const u = await ledger.getUser()
```

Comment:
I believe we fetch the user multiple times now. We already fetched it once to get the actual party and it looks like now you fetched again to get the user via `getPrimaryParty`. Maybe we can merge the two and only fetch once?

Reply:
That's true. I think the best way would be to store the username in the credentials and then expose a `useUser` hook in `daml-react` analogous to the `useParty` hook. I'd prefer adding new react hooks in a separate PR as I tried and failed before. Also not quite sure how to get the username in the case of Damlhub.